test/studies/shootout/meteor/kbrady/meteor-parallel.chpl

```diff
@@ -532,25 +537,22 @@ module meteor {
 
 
   proc solve() {
-    forall piece in 0..9:uint(8) {
+    forall piece in piecesDom do
       solve_helper(piece);
-    }
   }
 
-  proc solve_linear(in depth: int, in cell: int, in board: uint,
-      in avail: uint(16), solNums: [] uint(8), solMasks: [] uint) {
-    var pieceNoMask: uint(16);
-    var maxRots: int;
-    var pieceMask: uint;
+  proc solve_linear(in depth, in cell, in board,
+      in avail, solNums, solMasks) {
+    var pieceNoMask, maxRots, pieceMask: int;
 
     if solutionCount.read() >= maxSolutions then
       return;
 
     while (board & (1 << cell)) do
       cell += 1;
 
-    for piece in 0..9:uint(8) {
-      pieceNoMask = 1:uint(16) << piece;
+    for piece in piecesDom {
+      pieceNoMask = 1 << piece;
       if !((avail & pieceNoMask):bool) {
```

Comment:
I would guess not. 

As I understand, what this currently does is check if (avail bit-AND pieceNoMask) has any bits in it. Let's say they were int8s, and that they do have bits left, so they have some anded value 00110001 (chosen for the sake of explanation, not because it actually occurs). If cast to a bool, this means 'true'. Negated, 'true' means 'false'. If not cast to a bool, this means 00110001, which negated means 11001110, which cast to a bool (for the if statement) then means 'true'.